gitindex: respect flag -submodule

stefanhengl opened the pull request:
With this change we don't set `RepoCache` if `-submodule=false`;

It turns out that we didn't respect the submodule flag, IE submodules was enabled by default.
This caused issues, e.g. if a repository had an invalid .gitmodules file.

I also wrapped some of the high-level calls that gitindex does to generate better error
messages and speed-up future debugging.

stefanhengl committed:
wrap errors

This adds a bit of context to some errors to make it easier to debug
issues during indexing.

stefanhengl committed:
ignore but log errors from parsing .gitmodules

keegancsmith commented on gitindex/index.go:
```diff
@@ -403,17 +403,18 @@ func IndexGitRepo(opts Options) error {
 		files, subVersions, err := TreeToFiles(repo, tree, opts.BuildOptions.RepositoryDescription.URL, repoCache)
 		if err != nil {
-			return err
+			return fmt.Errorf("TreeToFiles: %w", err)
```
nit: idiomatic go has error messages not start with a capital

keegancsmith commented on gitindex/tree.go:
```diff
@@ -99,7 +99,7 @@ func TreeToFiles(r *git.Repository, t *object.Tree,
 
 	if err := rw.parseModuleMap(t); err != nil {
-		return nil, nil, err
+		log.Printf("parseModuleMap: %s, ignoring error", err)
```
I think somewhere we disable submodule support in our zoekt invocation. Can we maybe just skip these steps in that case?

stefanhengl committed:
mv submodules submoduleMap

renaming because I want to use the name submodules for something else in
the next commit

stefanhengl committed:
respect flag -submodules

stefanhengl committed:
Revert "respect flag -submodules"

This reverts commit 0249af78d7a3a27196604e17862697fb692db18e.

stefanhengl committed:
Revert "mv submodules submoduleMap"

This reverts commit 2cc9a8c62aa1140fc16de7b88a8af12b90203024.

stefanhengl committed:
don't set repoCache if !submodules

stefanhengl committed:
%s -> %w
